atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTopologyValidatorTest.java

```diff
@@ -0,0 +1,304 @@
+                .isNotEmpty();
+        assertThat(metrics.validationFailures().getCount()).isEqualTo(1);
+        assertThat(metrics.validationLatency().getCount()).isEqualTo(1);
+    }
+
+    @Test
+    public void returnsEmptyWhenGivenNoServers() {
+        assertThat(validator.getNewHostsWithInconsistentTopologies(ImmutableSet.of(), ImmutableMap.of()))
+                .isEmpty();
+    }
+
+    @Test
+    public void throwsWhenAllHostsDoesNotContainNewHosts() {
+        assertThatThrownBy(() -> validator.getNewHostsWithInconsistentTopologies(
+                        ImmutableSet.of(createCassandraServer("foo")), ImmutableMap.of()))
+                .isInstanceOf(SafeIllegalArgumentException.class);
+    }
+
+    @Test
+    public void returnsEmptyWhenOnlyNewServersAndAllHaveSameHostIds() {
+        Map<CassandraServer, CassandraClientPoolingContainer> allHosts = setupHosts(NEW_HOSTS);
+        setHostIds(allHosts.values(), DEFAULT_RESULT);
+        assertThat(validator.getNewHostsWithInconsistentTopologies(allHosts.keySet(), allHosts))
+                .isEmpty();
+    }
+
+    @Test
+    public void returnsNewHostsWhenOnlyNewServersAndOneMismatch() {
+        Map<CassandraServer, CassandraClientPoolingContainer> allHosts = setupHosts(NEW_HOSTS);
+        setHostIds(
+                filterContainers(allHosts, server -> !server.equals(NEW_HOST_ONE)),
+                HostIdResult.success(ImmutableSet.of("uuid1", "uuid2")));
+        setHostIds(
+                filterContainers(allHosts, NEW_HOST_ONE::equalsIgnoreCase),
+                HostIdResult.success(ImmutableSet.of("uuid3", "uuid2")));
+        assertThat(validator.getNewHostsWithInconsistentTopologies(allHosts.keySet(), allHosts))
+                .containsExactlyElementsOf(allHosts.keySet());
+    }
+
+    @Test
+    public void returnsNewHostsWhenNewServersMismatchOldServers() {
+        Map<CassandraServer, CassandraClientPoolingContainer> allHosts = setupHosts(ALL_HOSTS);
+        Set<CassandraServer> newServers = filterServers(allHosts, NEW_HOSTS::contains);
+        setHostIds(
+                filterContainers(allHosts, NEW_HOSTS::contains),
+                HostIdResult.success(ImmutableSet.of("uuid1", "uuid2")));
+        setHostIds(
+                filterContainers(allHosts, OLD_HOSTS::contains),
+                HostIdResult.success(ImmutableSet.of("uuid3", "uuid2")));
+        assertThat(validator.getNewHostsWithInconsistentTopologies(newServers, allHosts))
+                .containsExactlyElementsOf(newServers);
+    }
+
+    @Test
+    public void validateNewlyAddedHostsReturnsOnlyMismatchingNewHosts() {
+        Predicate<String> badHostFilter = NEW_HOST_ONE::equalsIgnoreCase;
+        Map<CassandraServer, CassandraClientPoolingContainer> allHosts = setupHosts(ALL_HOSTS);
+        Set<CassandraServer> newServers = filterServers(allHosts, NEW_HOSTS::contains);
+        Set<CassandraServer> badNewHosts = filterServers(allHosts, badHostFilter);
```

Comment:
nit: would suggest either moving these to be after line 159, or inlining them.

In general, variables should be declared as close to their point of usage as reasonable. I also think these are self-descriptive enough to be inlined, though happy to have these remain separate declarations if you'd prefer.

Reply:
yup can move, I prefer to keep variable names just b/c it makes reading the assert much more easier